### PR TITLE
Enhance app messaging

### PR DESCRIPTION
**Pull request by kildre**
Added better error handling and messaging to create app form.
## Reference Issues or PRs
https://gitlab.jatic.net/jatic/team-metrostar/t-e-platform/-/issues/643

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [x] Did you add new tests?

#### Text styling

- [x] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [x] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [x] If the content were to be read as plain text, it still makes sense, and no information is missing.

## Any other comments?
image
image

### CODE REVIEW

File: ui/src/components/app-form/app-form.tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
+import ErrorIcon from '@mui/icons-material/Error';
```
Comment (jbouder) [inline ui/src/components/app-form/app-form.tsx:1]: This should probably be this: `import ErrorRoundedIcon from '@mui/icons-material/ErrorRounded';`

File: ui/src/components/app-form/app-form.tsx [hunk 2]
```diff
@@ -181,33 +185,24 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
...
+        description: data.description,
+        framework: data.framework,
+        thumbnail: data.thumbnail,
+        filepath: data.filepath,
+        conda_env: data.conda_env,
         env: getFriendlyEnvironmentVariables(variables),
-        custom_command,
-        profile,
+        custom_command: data.custom_command,
+        profile: data.profile,
```
Comment (jbouder) [inline ui/src/components/app-form/app-form.tsx:205]: Why were these changed to using the data object?

File: ui/src/components/app-form/app-form.tsx [hunk 3]
```diff
@@ -224,14 +219,14 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
...
+          description: data.description || '',
+          framework: data.framework,
+          thumbnail: data.thumbnail || '',
+          filepath: data.filepath || '',
+          conda_env: data.conda_env || '',
           env: getFriendlyEnvironmentVariables(variables),
-          custom_command: custom_command || '',
-          profile: profile || '',
+          custom_command: data.custom_command || '',
+          profile: data.profile || '',
```
Comment (jbouder) [inline ui/src/components/app-form/app-form.tsx:229]: Same comment as above

File: ui/src/components/app-form/app-form.tsx [hunk 4]
```diff
@@ -376,33 +371,91 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
...
-          rules={REQUIRED_FORM_FIELDS_RULES}
-          // eslint-disable-next-line @typescript-eslint/no-unused-vars
-          render={({ field: { ref: _, ...field } }) => (
-            <FormControl>
+          rules={{ required: true }}
+          render={({ field: { ref, ...field } }) => (
+            <FormControl error={!!errors.display_name}>
+              {errors.display_name && (
+                <Box
+                  // id="display_name"
```
Comment (jbouder) [inline ui/src/components/app-form/app-form.tsx:432]: Remove this

File: ui/src/components/app-form/app-form.tsx [hunk 5]
```diff
@@ -486,45 +539,154 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
...
-                error={errors.framework?.message ? true : false}
+                error={!!errors.framework}
+                displayEmpty
+                inputProps={{ 'aria-label': 'Select framework' }}
               >
+                {errors.framework && (
+                  <MenuItem
+                    value=""
+                    disabled
+                    sx={{ color: 'limegreen' }}
```
Comment (jbouder) [inline ui/src/components/app-form/app-form.tsx:583]: Is this supposed to be lime green?

File: ui/src/components/app-sharing/app-sharing.tsx [hunk 6]
```diff
@@ -102,9 +100,9 @@ export const AppSharing = ({
   setIsPublic,
 }: AppSharingProps): React.ReactElement => {
   const [currentUser] = useRecoilState<UserState | undefined>(defaultUser);
-  const [message] = useState(
-    'Warning: adding individuals or groups will allow others to access this app.',
-  );
+  // const [message] = useState(
+  //   'Warning: adding individuals or groups will allow others to access this app.',
+  // );
```
Comment (jbouder) [inline ui/src/components/app-sharing/app-sharing.tsx:105]: Remove this

File: ui/src/components/app-sharing/app-sharing.tsx [hunk 7]
```diff
@@ -187,16 +185,6 @@ export const AppSharing = ({
...
-            <Item>
-              <Alert
-                id="sharing-notification"
-                severity="warning"
-                icon={<WarningRoundedIcon sx={{ color: '#EAB54E' }} />}
-                sx={{ mb: '16px' }}
-              >
-                {message}
-              </Alert>
-            </Item>
```
Comment (jbouder) [inline ui/src/components/app-sharing/app-sharing.tsx:199]: Why was this removed?

File: ui/src/components/custom-label/custom-label.tsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+  style?: React.CSSProperties & { requiredColor?: string };
+}
+
+export const CustomLabel = ({
+  label,
+  required,
+  className,
+  style,
+}: CustomLabelProps): React.ReactElement => {
+  const classes = classnames('custom-label', className);
```
Comment (jbouder) [inline ui/src/components/custom-label/custom-label.tsx:16]: Is this doing anything? I don't see any css added to this component

File: ui/src/components/thumbnail/thumbnail.tsx [hunk 9]
```diff
@@ -119,13 +121,44 @@ export const Thumbnail = ({
...
+        width: '100%',
         border: 'none',
       }}
     >
+      {error ? (
+        <Box
+          sx={{
+            display: 'flex',
+            alignItems: 'center',
+            // backgroundColor: theme.palette.error.light,
```
Comment (jbouder) [inline ui/src/components/thumbnail/thumbnail.tsx:135]: Remove this

**Review by jbouder — CHANGES_REQUESTED** [review]

File: ui/src/components/app-form/app-form.tsx (hunk 1 above)
Comment (kildre) [reply ui/src/components/app-form/app-form.tsx:1]: Replaced
image
image

File: ui/src/components/app-sharing/app-sharing.tsx (hunk 7 above)
Comment (kildre) [reply ui/src/components/app-sharing/app-sharing.tsx:199]: It was one of the requirements in the definition of done.
image
https://gitlab.jatic.net/jatic/team-metrostar/t-e-platform/-/issues/643

File: ui/src/components/app-form/app-form.tsx (hunk 3 above)
Comment (kildre) [reply ui/src/components/app-form/app-form.tsx:229]: Not sure where that's coming from. This is what I have:
image

Comment (jbouder) [reply ui/src/components/app-form/app-form.tsx:229]: I'm referring to using the data object, as opposed to just the individual values.

File: ui/src/components/app-sharing/app-sharing.tsx (hunk 6 above)
Comment (kildre) [reply ui/src/components/app-sharing/app-sharing.tsx:105]: Removed 'Warning:' and put back in per conversation

File: ui/src/components/app-form/app-form.tsx (hunk 2 above)
Comment (kildre) [reply ui/src/components/app-form/app-form.tsx:205]: My AI assistant suggested that I use data, so I did. Reverted back to original

image
image

File: ui/src/components/app-form/app-form.tsx (hunk 5 above)
Comment (kildre) [reply ui/src/components/app-form/app-form.tsx:583]: No. Was looking for an obvious change in color for testing.

File: ui/src/components/custom-label/custom-label.tsx (hunk 8 above)
Comment (kildre) [reply ui/src/components/custom-label/custom-label.tsx:16]: Not at the moment, removed.

File: ui/src/components/thumbnail/thumbnail.tsx (hunk 9 above)
Comment (kildre) [reply ui/src/components/thumbnail/thumbnail.tsx]: Removed

File: ui/src/components/app-sharing/app-sharing.tsx (hunk 7 above)
Comment (kildre) [reply ui/src/components/app-sharing/app-sharing.tsx]: Added back in removed the word "Warning:"

**Review by jbouder — APPROVED** [review]
👍
